src/engine/init.c

```diff
@@ -870,6 +870,19 @@ Options:\n\
 		dss_socket_dir, dss_nvme_conf, dss_instance_idx);
 }
 
+static int arg_strtoul(const char *str, unsigned int *value)
+{
+	char *ptr_parse_end = NULL;
+
+	*value = strtoul(str, &ptr_parse_end, 0);
+	if (ptr_parse_end && *ptr_parse_end != '\0') {
+		printf("%s is not a valid numeric value.\n", str);
```

Comment:
It's better to print error message on the caller to explicitly specify which argument is wrong.